pancancer_utilities/data_utilities.py

```diff
@@ -252,4 +247,41 @@ def summarize_results(results, gene, holdout_cancer_type, signal, z_dim,
 
     return metrics_out_, roc_df_, pr_df_
 
+def subset_by_mad(X_train_df, X_test_df, gene_features, subset_mad_genes, verbose=False):
+    """Subset features by mean absolute deviation.
+
+    Takes the top subset_mad_genes genes (sorted in descending order),
+    calculated on the training set.
+
+    Arguments
+    ---------
+    X_train_df: training data, samples x genes
+    X_test_df: test data, samples x genes
+    gene_features: numpy array of gene features
+    subset_mad_genes (int): number of genes to take
+
+    Returns
+    -------
+    (train_df, test_df, gene_features) datasets with filtered features
+    """
+    if verbose:
+        print('Taking subset of gene features', file=sys.stderr)
+
+    mad_genes_df = pd.DataFrame(
+            X_train_df.loc[:, gene_features].mad(axis=0).sort_values(ascending=False)
+    ).reset_index()
```

Comment:
It should not be necessary to create another DataFrame I think.
```suggestion
mad_genes_df = X_train_df.loc[:, gene_features].mad(axis=0).sort_values(ascending=False).reset_index()
```